pkg/operator/rgw/rgw.go

```diff
@@ -89,8 +110,141 @@ func (c *Cluster) Start() error {
 	return nil
 }
 
+type idType struct {
+	ID string `json:"id"`
+}
+
+func (c *Cluster) createRealm(serviceIP string) error {
+	realmArg := fmt.Sprintf("--rgw-realm=%s", c.Config.Name)
+	zonegroupArg := fmt.Sprintf("--rgw-zonegroup=%s", c.Config.Name)
+	zoneArg := fmt.Sprintf("--rgw-zone=%s", c.Config.Name)
+	endpointArg := fmt.Sprintf("--endpoints=%s:%d", serviceIP, c.Config.Port)
+	updatePeriod := false
+
+	// create the realm if it doesn't exist yet
+	output, err := c.runRGWCommand("realm", "get", realmArg)
+	if err != nil {
+		updatePeriod = true
+		output, err = c.runRGWCommand("realm", "create", realmArg)
+		if err != nil {
+			return fmt.Errorf("failed to create rgw realm %s. %+v", c.Config.Name, err)
+		}
+	}
+
+	realmID, err := decodeID(output)
+	if err != nil {
+		return fmt.Errorf("failed to parse realm id. %+v", err)
+	}
+
+	// create the zonegroup if it doesn't exist yet
+	output, err = c.runRGWCommand("zonegroup", "get", zonegroupArg, realmArg)
+	if err != nil {
+		updatePeriod = true
+		output, err = c.runRGWCommand("zonegroup", "create", "--master", zonegroupArg, realmArg, endpointArg)
+		if err != nil {
+			return fmt.Errorf("failed to create rgw zonegroup for %s. %+v", c.Config.Name, err)
+		}
+	}
+
+	zoneGroupID, err := decodeID(output)
+	if err != nil {
+		return fmt.Errorf("failed to parse realm id. %+v", err)
+	}
+
+	// create the zone if it doesn't exist yet
+	output, err = c.runRGWCommand("zone", "get", zoneArg, zonegroupArg, realmArg)
+	if err != nil {
+		updatePeriod = true
+		output, err = c.runRGWCommand("zone", "create", "--master", endpointArg, zoneArg, zonegroupArg, realmArg)
+		if err != nil {
+			return fmt.Errorf("failed to create rgw zonegroup for %s. %+v", c.Config.Name, err)
+		}
+	}
+	zoneID, err := decodeID(output)
+	if err != nil {
+		return fmt.Errorf("failed to parse zone id. %+v", err)
+	}
+
+	if updatePeriod {
+		_, err = c.runRGWCommand("period", "update", "--commit")
```

Comment:
what does update period do?  we are doing it conditionally, perhaps a comment on why and what it does could help

Reply:
will add a comment. It's a way of updating an epoch so interested parties can be aware when a realm or zone is created